crates/conf/src/macros.rs

```diff
@@ -0,0 +1,146 @@
+use anyhow::Error;
+use thiserror::Error as ErrorDerive;
+
+#[derive(Debug, ErrorDerive)]
+pub enum ConfigLoadError {
+    #[error("Failed to load configuration: {0}")]
+    Error(Error),
+    #[error("Configuration check failed: {0:?}")]
+    CheckErrors(Vec<(String, Error)>),
```

Comment:
I would enforce a specific type for each error. It would make it simpler. Ideally the type should implement Display (so we do not have to use debug printing). Maybe a String?

Reply:
What do you mean by this? types for `CheckErrors` or just more `Error` types in the enum?

Reply:
I mean that as of now, there is no convincing reason (AFAIK) to support generic field errors. So I would change this to `CheckErrors(Vec<(String, String)>)` and document that this is a tuples with field and error description respectively.